apps/src/applab/DesignToggleRow.jsx

```diff
@@ -0,0 +1,69 @@
+var React = require('react');
+var msg = require('../locale');
+
+var Mode = {
+  CODE: 'CODE',
+  DESIGN: 'DESIGN'
+};
+
+module.exports = React.createClass({
+  propTypes: {
+    screens: React.PropTypes.array.isRequired,
+    onDesignModeButton: React.PropTypes.func.isRequired,
+    onCodeModeButton: React.PropTypes.func.isRequired
+  },
+
+
+  getInitialState: function () {
+    return {
+      mode: Mode.CODE
+    };
+  },
+
+  handleModeToggle: function () {
+    var newMode;
```

Comment:
oops, you can leave it this way, or you could put the setState call inline. either way is fine.